hooks/pre-commit

```diff
@@ -1,25 +1,27 @@
-#!/bin/bash
-if clang-format --version | grep -q 'version 11\.'; then
-   CLANG_FORMAT_EXECUTABLE="clang-format"
-else
-   CLANG_FORMAT_EXECUTABLE="clang-format-11"
-fi
+#!/bin/sh
+
+name=""
+version="14.0.0"
 
-if ! command -v $CLANG_FORMAT_EXECUTABLE &> /dev/null
-then
-    echo $CLANG_FORMAT_EXECUTABLE does not exist, make sure to install it
-    exit 1
+for file in $(find $(echo "$PATH" | sed 's/:/ /g') -maxdepth 1 -type f -executable -name 'git-clang-format*'); do
+  cur_name="$(basename "$file" | sed 's/^git-//')"
+  cur_version="$("$cur_name" --version | awk '{print $3}')"
+
+  if [ "$(printf '%s\n' "$cur_version" "$version" | sort -V | tail -n 1)" = "$cur_version" ]; then
+    version="$cur_version"
+    name="$cur_name"
+  fi
+done
```

Comment:
Wouldn't just `clang-format` always be the highest available version?

Reply:
No, in the case of Ubuntu 20.04, the version of `clang-format` is 10, even though `clang-format-12` is available in the repositories.